kms/asymmetric/samples_test.go

```diff
@@ -0,0 +1,160 @@
+package main
+
+import (
+	"crypto/ecdsa"
+	"crypto/rsa"
+	"github.com/GoogleCloudPlatform/golang-samples/internal/testutil"
+	"golang.org/x/net/context"
+	"golang.org/x/oauth2/google"
+	"google.golang.org/api/cloudkms/v1"
+	"testing"
+)
+
+type TestVariables struct {
+	client         *cloudkms.Service
+	ctx            context.Context
+	message        string
+	rsaDecryptPath string
+	rsaSignPath    string
+	ecSignPath     string
+}
+
+func setup(t *testing.T) TestVariables {
+	tc := testutil.SystemTest(t)
+	projectId := tc.ProjectID
+	parent := "projects/" + projectId + "/locations/global"
+	keyRing := "kms-asymmetric-samples"
+
+	rsaDecrypt := parent + "/keyRings/" + keyRing + "/cryptoKeys/rsa-decrypt/cryptoKeyVersions/1"
+	rsaSign := parent + "/keyRings/" + keyRing + "/cryptoKeys/rsa-sign/cryptoKeyVersions/1"
+	ecSign := parent + "/keyRings/" + keyRing + "/cryptoKeys/ec-sign/cryptoKeyVersions/1"
+
+	message := "test message 123"
+
+	ctx := context.Background()
+	client, err := google.DefaultClient(ctx, cloudkms.CloudPlatformScope)
+	if err != nil {
+		t.Fatal(err)
+	}
+	kmsClient, err := cloudkms.New(client)
+	if err != nil {
+		t.Fatal(err)
+	}
+	//create cryptokeys
+	kmsClient.Projects.Locations.KeyRings.Create(parent, &cloudkms.KeyRing{}).KeyRingId(keyRing).Do()
+	kmsClient.Projects.Locations.KeyRings.CryptoKeys.Create(
+		parent+"/keyRings/"+keyRing, &cloudkms.CryptoKey{
+			Purpose: "ASYMMETRIC_DECRYPT",
+			VersionTemplate: &cloudkms.CryptoKeyVersionTemplate{
+				Algorithm: "RSA_DECRYPT_OAEP_2048_SHA256",
+			},
+		}).CryptoKeyId("rsa-decrypt").Do()
+	kmsClient.Projects.Locations.KeyRings.CryptoKeys.Create(
+		parent+"/keyRings/"+keyRing, &cloudkms.CryptoKey{
+			Purpose: "ASYMMETRIC_SIGN",
+			VersionTemplate: &cloudkms.CryptoKeyVersionTemplate{
+				Algorithm: "RSA_SIGN_PSS_2048_SHA256",
+			},
+		}).CryptoKeyId("rsa-sign").Do()
+	kmsClient.Projects.Locations.KeyRings.CryptoKeys.Create(
+		parent+"/keyRings/"+keyRing, &cloudkms.CryptoKey{
+			Purpose: "ASYMMETRIC_SIGN",
+			VersionTemplate: &cloudkms.CryptoKeyVersionTemplate{
+				Algorithm: "EC_SIGN_P224_SHA256",
+			},
+		}).CryptoKeyId("ec-sign").Do()
+
+	v := TestVariables{kmsClient, ctx, message, rsaDecrypt, rsaSign, ecSign}
+	return v
+}
+
+//test equality between two values
+func assertEqual(t *testing.T, a interface{}, b interface{}) {
+	if a != b {
+		t.Errorf("%s != %s", a, b)
+	}
+}
+
+func TestGetPublicKey(t *testing.T) {
+	v := setup(t)
+
+	rsaDecryptPub, err := getAsymmetricPublicKey(v.client, v.ctx, v.rsaDecryptPath)
+	if err != nil {
+		t.Fatal(err)
+	}
+	_, ok := rsaDecryptPub.(*rsa.PublicKey)
+	assertEqual(t, ok, true)
```

Comment:
please, no helper functions. Just write the code.

This `assertEqual` function loses all its context. I'd expect a test failure message like:

```
if _, ok := rsaDecryptPub.(*rsa.PublicKey); !ok {
   t.Errorf("type of rsaDecryptPub: got %T, want *rsa.PublicKey", rsaDecryptPub)
}
```

The current message would say: "false != true", which is not very helpful.